### ISSUE TITLE
Cassava Leaf Disease Classification

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#418]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Cassava Leaf Disease Classification</br>
:red_circle: **Aim** : The aim of this project is to identify the defected leaf using image processing methods.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/nirmalsankalana/cassava-leaf-disease-classification</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by VaishnaviMudaliar** [comment]
Hey! Hope you are doing well, I am interested in doing this project and would request you to assign this issue to me 😊 
Thanks a lot!

**Comment by abhisheks008** [comment]
Are you part of the SWOC event?
@VaishnaviMudaliar

**Comment by VaishnaviMudaliar** [comment]
Yes, I am a contributor at swoc

**Comment by abhisheks008** [comment]
Hi @VaishnaviMudaliar please share your approach for solving this issue, what the models you are planning to implement for this project, do let me know here.

**Comment by Aryan863** [comment]
Hey @abhisheks008  i am interested in this project. should i mention my approach?

**Comment by arijitde92** [comment]
Hi @abhisheks008 , I am interested in doing this project and would request you to assign this issue to me. Thank you. Below are the requested details-

Full name : Arijit De
GitHub Profile Link : [https://github.com/arijitde92](https://github.com/arijitde92)
Email ID : [arijitde2050@gmail.com](arijitde2050@gmail.com)
Participant ID (if applicable): N/A
Approach for this Project : 
1. Exploratory Data Analysis
First I will download the data and check for corrupt images and remove them if present. I will also check whether the dataset has class imbalance or not. I will provide some report about the number of samples in each class.
2. Data preprocessing
Will find out the mean and standard deviation of the images so that I can normalize the data.
3. Data Splitting
The dataset contains five classes and they are not already divided into train/test sets (as given in Kaggle). I will divide the dataset into training/testing sets in the ratio 80:20 (will experiement with other ratios and see which is best).
4. Data Augmentation
Since the dataset is class imbalanced, training any deep learning model will cause biasness for majority class. Hence I will do data augmentation (rotation, scaling, flipping) to increase the samples of minority classes in the training set so that all classes have the same number of images.
5. Model Training
I will train the images with four different models - EfficientNet, DenseNet, MobileNet and SqueezeNet. Some are heavy models while some are lightweight models. I will experiment and see how each model performs. Will also experiment with pre-trained versions of the models.
6. Model Hyper parameters
I will also experiment with the model hyperparameters like learning rate, batch size and optimizer to find the best model.
7. Testing
I will test the model on the test set and apart from accuracy, will determine the precision, recall, sensitivity and specificity of the models.

What is your participant role? SWOC 24 contributor.

**Comment by abhisheks008** [comment]
Issue assigned to you @arijitde92
